Source/Scene/ModelAnimationCollection.js

```diff
@@ -397,22 +409,34 @@ ModelAnimationCollection.prototype.update = function (frameState) {
     const duration = scheduledAnimation._duration;
     const stopTime = scheduledAnimation.stopTime;
 
+    const pastStartTime = JulianDate.lessThanOrEquals(startTime, sceneTime);
+    const pastStopTime =
+      defined(stopTime) && JulianDate.greaterThan(sceneTime, stopTime);
+
     // [0.0, 1.0] normalized local animation time
     let delta =
       duration !== 0.0
-        ? JulianDate.secondsDifference(sceneTime, startTime) / duration
+        ? scheduledAnimation.animationTime
+          ? scheduledAnimation.animationTime(duration)
```

Comment:
Can we separate these ternary operators so it's more readable? For example

```
let animationDelta= defined(scheduleAnimation.animationTime) ?
   scheduledAnimation.animationTime(duration)
   : JulianDate.secondsDifference(...);
let delta = duration !== 0.0 ? animationDelta : 0.0;
```

Reply:
I took a slightly different approach, because most animationTime functions are going to want to divide by duration, and the default operation *does* divide by duration (you omitted that in your rewrite, but it should have been `JulianDate.secondsDifference(...) / duration`.
Perhaps you meant to write
```
let delta = duration !== 0.0 ? animationDelta / duration : 0.0;
```
but then animationTime has to return something of the form `N * duration + m`, with `N` an integer and `m < duration`. I think its more natural for the animationTime function to return something of the form `N + m` with `N` integer, and `0 <= m < 1`.